### PR TITLE
Rules CRUD 

### PR DESCRIPTION
**Pull request by somnath77**
```
1. Added CRUD for rules
2. Added unit tests for same
```

### CODE REVIEW

**Review by rahulknojha — APPROVED** [review]

File: app/controllers/api/v1/admin/rules_controller.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            render_success(data: { rule: serialize_resource(rule, RuleSerializer) },
+                           message: I18n.t('create.success', model_name: Rule))
+          else
+            render_error(message: rule.errors.messages, status: 400)
+          end
+        end
+
+        def index
+          drive_id = Drive.find(params[:id])
+          rules = Rule.where(drive_id: drive_id)
```
Comment (vaibhav8186) [inline app/controllers/api/v1/admin/rules_controller.rb:22]: @somnath77  can we use association to find records.

File: app/controllers/api/v1/admin/rules_controller.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+          render_success(data: { rules: serialize_resource(rules, RuleSerializer) },
+                         message: I18n.t('index.success', model_name: Rule))
+        end
+
+        def update
+          rule = Rule.find(params[:id])
```
Comment (vaibhav8186) [inline app/controllers/api/v1/admin/rules_controller.rb:29]: @somnath77  can we add one object finder method in the callback.

File: app/controllers/api/v1/admin/rules_controller.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
```
Comment (vaibhav8186) [inline app/controllers/api/v1/admin/rules_controller.rb:4]: @somnath77   can we take it in one line - `Api::v1::Admin`

**Review by vaibhav8186 — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/admin/rules_controller.rb (hunk 3 above)
Comment (somnath77) [reply app/controllers/api/v1/admin/rules_controller.rb:4]: I did same initially, but rubocop auto corrects it as nested module form

File: app/controllers/api/v1/admin/rules_controller.rb (hunk 2 above)
Comment (somnath77) [reply app/controllers/api/v1/admin/rules_controller.rb:29]: Added

File: app/controllers/api/v1/admin/rules_controller.rb (hunk 1 above)
Comment (somnath77) [reply app/controllers/api/v1/admin/rules_controller.rb]: Done

**Review by vaibhav8186 — APPROVED** [review]

**Pull request merged**